animation_nodes/nodes/gpencil/replicate_gp_layer.py

```diff
@@ -0,0 +1,80 @@
+import bpy
+from bpy.props import *
+from mathutils import Matrix
+from ... data_structures import GPLayer, VirtualDoubleList
+from ... base_types import AnimationNode, VectorizedSocket
+
+transformationTypeItems = [
+    ("Matrix List", "Matrices", "", "NONE", 0),
+    ("Vector List", "Vectors", "", "NONE", 1)
+]
+
+class ReplicateGPLayerNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_ReplicateGPLayerNode"
+    bl_label = "Replicate GP Layer"
+
+    useLayerList: VectorizedSocket.newProperty()
+
+    transformationType: EnumProperty(name = "Transformation Type", default = "Matrix List",
+        items = transformationTypeItems, update = AnimationNode.refresh)
+
+    def create(self):
+        self.newInput(VectorizedSocket("GPLayer", "useLayerList",
+            ("Layer", "layer", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Layers", "layers")))
+        self.newInput(self.transformationType, "Transformations", "transformations")
+        self.newInput("Float List", "Offset Frame Numbers", "offsets", value = 0, hide = True)
```

Comment:
I don't think we should have offsets here. Users can utilize the Offset node for that.

Reply:
It is easy to offset _frame-number_ with the Offset node when we are replicating a single layer. However, if we are replicating two or more layers then it is difficult to offset _frame-number_ with Offset node. Here is the comparison,
**Offset with the Replicate Layer**,
![Screenshot from 2020-02-14 01-16-41](https://user-images.githubusercontent.com/30294746/74472381-da879b00-4ec7-11ea-95e7-873c57770755.png)

**Offset with the Offset node**,
![Screenshot from 2020-02-14 01-16-19](https://user-images.githubusercontent.com/30294746/74472429-ea9f7a80-4ec7-11ea-9849-505517bb8906.png)

Full scene,
![Screenshot from 2020-02-14 01-17-07](https://user-images.githubusercontent.com/30294746/74472463-fab75a00-4ec7-11ea-8ffc-d54529f9ab5e.png)

Reply:
I see. Ok. Rename the input to just *Offset*. Should it be a vectorizaed virtual list?

Reply:
Okay. I think yes.

Reply:
Omar, should we name it as _Offset Frame(s)_?

Reply:
Ok.